app/packages/core/src/components/Modal/Group/DynamicGroup/carousel/DynamicGroupsFlashlightWrapper.tsx

```diff
@@ -0,0 +1,302 @@
+import Flashlight, { Response } from "@fiftyone/flashlight";
+import { freeVideos, zoomAspectRatio } from "@fiftyone/looker";
+import * as foq from "@fiftyone/relay";
+import * as fos from "@fiftyone/state";
+import React, {
+  MutableRefObject,
+  useEffect,
+  useId,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useErrorHandler } from "react-error-boundary";
+import { usePaginationFragment } from "react-relay";
+import {
+  useRecoilCallback,
+  useRecoilValue,
+  useRecoilValueLoadable,
+} from "recoil";
+import { useGroupContext } from "../../GroupContextProvider";
+
+export const DYNAMIC_GROUPS_FLASHLIGHT_CONTAINER_ID =
+  "dynamic-groups-flashlight-container";
+export const DYNAMIC_GROUPS_FLASHLIGHT_ELEMENT_ID =
+  "dynamic-groups-flashlight-element";
+
+const process = (
+  next: MutableRefObject<number>,
+  store: fos.LookerStore<any>,
+  zoom: boolean,
+  edges: foq.paginateGroup_query$data["samples"]["edges"]
+) =>
+  edges.reduce((acc, { node }) => {
+    if (
+      node.__typename === "ImageSample" ||
+      node.__typename === "VideoSample"
+    ) {
+      let data: any = {
+        sample: node.sample,
+        aspectRatio: node.aspectRatio,
+        urls: Object.fromEntries(
+          node.urls.map(({ field, url }) => [field, url])
+        ),
+      };
+
+      if (node.__typename === "VideoSample") {
+        data = {
+          ...data,
+          frameRate: node.frameRate,
+          frameNumber: node.sample.frame_number,
+        };
+      }
+
+      store.samples.set(node.sample._id, data);
+      store.indices.set(next.current, node.sample._id);
+      next.current++;
+
+      return [
+        ...acc,
+        {
+          aspectRatio: zoom
+            ? zoomAspectRatio(node.sample as any, node.aspectRatio)
+            : node.aspectRatio,
+          id: (node.sample as any)._id as string,
+        },
+      ];
+    }
+
+    return acc;
+  }, []);
+
+export const DynamicGroupsFlashlightWrapper = () => {
+  const id = useId();
+  const pageCount = useRef(0);
+
+  const { groupByFieldValue } = useGroupContext();
+
+  const { data, hasNext, loadNext } = usePaginationFragment(
+    foq.paginateGroupPaginationFragment,
+    useRecoilValue(
+      fos.dynamicGroupPaginationFragment({
+        fieldOrExpression: groupByFieldValue!,
+      })
+    )
+  );
+
+  // todo: support pcd
```

Comment:
If samples itself is point-cloud do we handle that well? Or would it be an empty flashlight?